### DISCUSSION TITLE
Write on Script language about commons with other programming languages

### DISCUSSION THREAD

**Discussion by rrolla** [savonet/liquidsoap#2102] [Ideas]
For example:

Liquidsoap `unit` is the same as in most of the languages `void`
Liquidsoap `!` with in front of the exclamation mark is the same as in most of the languages `&` ampersand (get value by reference)
Liquidsoap function automatically return the last line in function as in most of the languages there is `return` keyword at end of function

Cannot understand what means that underscore `_` eg commonly see like this `-> _` 
Cannot understand what means that one tick in front of variable `'` eg commonly see like this `source('a)` 

Should be more, but this for a starting point.

Cheers

**Comment by toots** [comment]
Hi! Underscore is used to denote when a variable is not used. `'` is used to denote _universal_ types, that is types that can be anything. For instance, a function of type `('a, 'b) -> unit` takes two variables of any type and returns nothing. You can write it like this:
```
# def f(_,_) = () end;;
f : ('a, 'b) -> unit = <fun>
```

**Comment by rrolla** [comment]
As I understand from the documentation: 
https://www.liquidsoap.info/doc-2.0.3/language.html#lists

https://www.liquidsoap.info/doc-2.0.3/language.html#tuples

Lists and Tuples are basically arrays only difference is that lists can save the **same type**, but tuples can save **different types**?
 
### About lists
And to access the list, we can use like in javascript Destructuring assignment https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment
```
myList = [2, 5]
let [firstValue, secondValue] = myList

# so print these variables
print(firstValue) 
# will print 2
print(secondValue) 
# will print 5
```

Is there some other way to access list element for example like this `myList[0]`?

### About tuples

Looks like here also works Destructuring assignment

```
myTuple = (3, "hello")
let (firstValue, secondValue) = myTuple
# so print these variables
print(firstValue) 
# will print 3
print(secondValue) 
# will print hello
```

Is there some other way to access the tuple element?

**Reply by toots** [reply]
This is all correct. Also, lists have unspecified number of elements while tuples have fixed number of elements.

`list.nth` can be used to access list elements: https://www.liquidsoap.info/doc-2.0.3/reference.html#list.nth and also` list.hd` for the first one. `list.tail` returns a list without its first element.

Tuples are more tricky because there is no generic type, each one has a different type and pre-defined number of elements. For pairs (tuples made of two elements), we have `fst` and `snd`.

**Reply by rrolla** [reply]
Thanks for the clarification @toots it's getting more transparent how it works.
